common/expression/src/function.rs

```diff
@@ -477,157 +494,160 @@ fn erase_calc_domain_generic_2_arg<I1: ArgType, I2: ArgType, O: ArgType>(
 }
 
 fn erase_function_generic_0_arg<O: ArgType>(
-    func: impl Fn(&GenericMap) -> Value<O>,
-) -> impl Fn(&[ValueRef<AnyType>], &GenericMap) -> Value<AnyType> {
-    move |_args, generics| func(generics).upcast()
+    func: impl Fn(&GenericMap) -> Result<Value<O>, String>,
+) -> impl Fn(&[ValueRef<AnyType>], &GenericMap) -> Result<Value<AnyType>, String> {
+    move |_args, generics| func(generics).map(Value::upcast)
 }
 
 fn erase_function_generic_1_arg<I1: ArgType, O: ArgType>(
-    func: impl Fn(ValueRef<I1>, &GenericMap) -> Value<O>,
-) -> impl Fn(&[ValueRef<AnyType>], &GenericMap) -> Value<AnyType> {
+    func: impl Fn(ValueRef<I1>, &GenericMap) -> Result<Value<O>, String>,
```

Comment:
If we always return `Ok`, will this prevent the Auto-Vectorized optimization?

Reply:
There are four ways to register a function overload: `register_n_arg`, `register_with_writer_n_arg`, `register_n_arg_core`, and `register_factory`. They have different features as described in the table:

|                            | Auto Vectorization | Auto Null Passthrough | Auto Downcast | Access Output Column Builder | Throw Runtime Error | Varidic |
|----------------------------|--------------------|-----------------------|---------------|------------------------------|---------------------|---------|
| register_n_arg             | ✅                  | ✅                     | ✅             | ❌                            | ❌                   | ❌       |
| register_with_writer_n_arg | ✅                  | ✅                     | ✅             | ✅                            | ✅                   | ❌       |
| register_n_arg_core        | ❌                  | ❌                     | ❌             | ✅                            | ✅                   | ❌       |
| register_factory           | ❌                  | ❌                     | ❌             | ✅                            | ✅                   | ✅       |

Reply:
Ok, let's keep this table in mind.

Reply:
> They have different features as described in the table:

Do we have this table in docs or code comments?